psydac/api/essential_bc.py

```diff
@@ -115,50 +119,81 @@ def apply_essential_bc_stencil(a, *, axis, ext, order, identity=False):
         pass
 
 #==============================================================================
-def apply_essential_bc_BlockMatrix(a, bc, identity=False):
-    """ Apply homogeneous dirichlet boundary conditions in nD """
+def apply_essential_bc_BlockMatrix(a, bc, *, identity=False, is_broken=True):
+    """ Apply homogeneous dirichlet boundary conditions in nD.
+        is_broken is used to identify if we are in a multipatch setting, where we assume
+        that the domain and codomain of each block of the BlockMatrix corresponds to a single patch.
+
+    Parameters
+    ----------
+    a : BlockMatrix
+        The BlockMatrix that will be modified.
+ 
+    bc: Sympde.expr.equation.BasicBoundaryCondition
+        The boundary condition type that will be applied to a.
+
+    is_broken: bool
+        Set to True if we are in a multipatch setting and False otherwise.
+    """
 
     assert isinstance(a, BlockMatrix)
     keys = list(a._blocks.keys())
```

Comment:
Instead of accessing the `_blocks` private member, could we not add a property to `BlockLinearOperator` which does what you want here? This is the fastest method on my machine:

```python
@property
def nonzero_block_indices(self):
    """
    Tuple of (i, j) pairs which identify the non-zero blocks:
    i is the row index, j is the column index.
    """
    return tuple(self._block)
```